server/src/main/java/io/deephaven/server/table/stats/CharacterChunkedStats.java

```diff
@@ -0,0 +1,112 @@
+
+            while (rsIt.hasMore()) {
+                // Grab up to the next CHUNK_SIZE rows
+                final RowSequence nextKeys = rsIt.getNextRowSequenceWithLength(CHUNK_SIZE);
+
+                final CharChunk<? extends Values> chunk =
+                        (usePrev ? columnSource.getPrevChunk(getContext, nextKeys)
+                                : columnSource.getChunk(getContext, nextKeys)).asCharChunk();
+                final int chunkSize = chunk.size();
+                for (int ii = 0; ii < chunkSize; ii++) {
+                    final char val = chunk.get(ii);
+
+                    if (val == QueryConstants.NULL_CHAR) {
+                        continue;
+                    }
+
+                    count++;
+
+                    if (useSet) {
+                        uniqueValues.add(val);
+                    } else if (uniqueCount > maxUniqueToCollect) {
+                        // we no longer need to track counts for these items; fall back to a Set to get at least a count
+                        uniqueValues.addAll(countValues.keySet());
+                        countValues.clear();
+                        uniqueValues.add(val);
+                        useSet = true;
+                    } else if (countValues.adjustOrPutValue(val, 1, 1) == 1) {
+                        uniqueCount++;
+                    }
+                }
+            }
+
+            final int numUnique;
+            String[] uniqueKeys;
+            long[] uniqueCounts;
+            if (useSet) {
+                numUnique = uniqueValues.size();
+                uniqueKeys = CollectionUtil.ZERO_LENGTH_STRING_ARRAY;
+                uniqueCounts = CollectionUtil.ZERO_LENGTH_LONG_ARRAY;
+            } else {
+                numUnique = countValues.size();
+                List<Map.Entry<String, Long>> sorted = new ArrayList<>();
+
+                countValues.forEachEntry((o, c) -> {
+                    sorted.add(Map.entry(Objects.toString(o), c));
+                    return true;
+                });
+                sorted.sort(Map.Entry.comparingByValue());
+
+                int resultCount = Math.min(maxUniqueToDisplay, sorted.size());
+                uniqueKeys = new String[resultCount];
+                uniqueCounts = new long[resultCount];
+                Iterator<Map.Entry<String, Long>> iter = sorted.iterator();
+                for (int i = 0; i < resultCount && iter.hasNext(); i++) {
+                    Map.Entry<String, Long> entry = iter.next();
+                    uniqueKeys[i] = entry.getKey();
+                    uniqueCounts[i] = entry.getValue();
+                }
+            }
+
+            return TableTools.newTable(
+                    TableTools.longCol("COUNT", count),
+                    TableTools.longCol("SIZE", rowSet.size()),
+                    TableTools.intCol("UNIQUE_VALUES", numUnique),
+                    new ColumnHolder<>("UNIQUE_KEYS", String[].class, String.class, false, uniqueKeys),
```

Comment:
Same question... can we skip these when empty? What even *happens* if we create these mismatched column sizes?

Reply:
Skipping when we don't populate makes more sense, the client can handle that case.

What happens if we create this mismatched intentionally? The client will have an error when it tries to use them... but at least with the current code, how is that possible?

Is this like asking what would happen if you grouped two columns on a table and they ended up with different sizes...?

Reply:
Scratch the size-mismatch question, that was reader-error.